src/rdkafka_cgrp.c

```diff
@@ -2434,6 +2435,280 @@ static void rd_kafka_cgrp_offset_commit_tmr_cb (rd_kafka_timers_t *rkts,
+         * making any changes to it to so the operation doesn't partially
+         * succeed. */
+        for (i = 0; rkcg->rkcg_assignment && i < partitions->cnt; i++) {
+                const rd_kafka_topic_partition_t *rktpar = &partitions->elems[i];
+                if (rd_kafka_topic_partition_list_find(rkcg->rkcg_assignment,
+                                                       rktpar->topic,
+                                                       rktpar->partition)) {
+                        return rd_kafka_error_new(
+                                RD_KAFKA_RESP_ERR__INVALID_ARG,
+                                "%s [%"PRId32"] is already part of the "
+                                "current assignment",
+                                rktpar->topic, rktpar->partition);
+                }
+        }
+
+        if (partitions->cnt == 0) {
+                rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER,
+                        "ASSIGN", "Group \"%s\": adding 0 partitions to "
+                        "existing assignment of %d partitions in join state "
+                        "%s (nothing to do)", rkcg->rkcg_group_id->str,
+                        !rkcg->rkcg_assignment ? 0
+                        : rkcg->rkcg_assignment->cnt,
+                        rd_kafka_cgrp_join_state_names[rkcg->rkcg_join_state]);
+                return NULL;
+        }
+
+        rd_kafka_cgrp_version_new_barrier(rkcg);
+
+        rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER, "ASSIGN",
+                "Group \"%s\": adding %d partitions to existing assignment "
+                "of %d partitions in join state %s (v%"PRId32")",
+                rkcg->rkcg_group_id->str, partitions->cnt,
+                !rkcg->rkcg_assignment ? 0 : rkcg->rkcg_assignment->cnt,
+                rd_kafka_cgrp_join_state_names[rkcg->rkcg_join_state],
+                rkcg->rkcg_version);
+
+        if (!rkcg->rkcg_assignment)
+                rkcg->rkcg_assignment = rd_kafka_topic_partition_list_new(
+                                                partitions->cnt);
+
+        /* Add and mark partition(s) as desired */
+        for (i = 0; i < partitions->cnt; i++) {
+                rd_kafka_toppar_t *rktp;
+                const rd_kafka_topic_partition_t *rktpar =
+                        &partitions->elems[i];
+
+                /* This marks the partition as desired if it's not already. */
+                rktp = rd_kafka_toppar_get2(rkcg->rkcg_rk,
+                                            rktpar->topic,
+                                            rktpar->partition,
+                                            0/*no-ua*/, 1/*create-on-miss*/);
+
+                if (!rktp)
+                        /* LOG_ERR already emitted by rd_kafka_toppar_get2 */
+                        continue;
+
+                rd_kafka_toppar_keep(rktp);
+
+                partitions->elems[i]._private = rktp;
+                if (!rd_kafka_topic_partition_list_add0(
```

Comment:
This function can't return NULL.

Reply:
how about that. i have no idea why i was checking for that.